### PR TITLE
Point and polygon buffers

### PR DESCRIPTION
**Pull request by turban**
Fixes: https://jira.dhis2.org/browse/DHIS2-6386

This PR adds support for point and polygon buffers.

### CODE REVIEW

File: src/layers/Buffer.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                ...feature,
+                geometry: turfBuffer(feature, options.buffer / 1000).geometry, // km
+            } : feature
+        );
+
+        GeoJson.prototype.initialize.call(this, options);
+    },
+
+    pointToLayer(geojson, latlng) {
+        this.options.style.pane = this.options.pane;
```
Comment (amcgee) [inline src/layers/Buffer.js]: Should `this.options.pane` be initialized somewhere?  Maybe it's in the GeoJson parent class, checking

File: src/layers/Buffer.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        // Draw points as circles using buffer as radius
+        return L.circle(latlng, {
+            ...this.options.style,
+            radius: this.options.buffer,
```
Comment (amcgee) [inline src/layers/Buffer.js]: Same with this one?

File: src/layers/FeatureGroup.js [hunk 3]
```diff
@@ -19,9 +19,9 @@ export const FeatureGroup = L.FeatureGroup.extend({
         const { buffer, bufferStyle, data, pane, opacity, isVisible } = this.options;
 
         if (buffer) {
-            const bufferLayer = new Circles({
+            const bufferLayer = new Buffer({
                 pane: `${pane}-buffer`,
-                radius: buffer,
+                buffer,
```
Comment (amcgee) [inline src/layers/FeatureGroup.js]: Ah I see.  I guess this is the only place this is used?

**Review by amcgee — APPROVED** [review]
Looks good unless we want to check the input to `Buffer.initialize`

File: src/layers/FeatureGroup.js (hunk 3 above)
Comment (turban) [reply src/layers/FeatureGroup.js]: Yes it is.

File: src/layers/Buffer.js (hunk 1 above)
Comment (turban) [reply src/layers/Buffer.js]: Added defult pane value.

File: src/layers/Buffer.js (hunk 2 above)
Comment (turban) [reply src/layers/Buffer.js]: Added defult buffer value (ranamed from radius).

**Comment by turban** [comment]
Fix included in @dhis2/gis-api@32.0.15

**Pull request merged**